mos_tests/nova/nova_test.py

```diff
@@ -601,3 +606,80 @@ def test_boot_instance_from_volume_bigger_than_flavor_size(self):
         # Check that instance is reachable
         ping = common_functions.ping_command(self.floating_ip.ip)
         self.assertTrue(ping, "Instance after creation is not reachable")
+
+
+class TestNovaDeferredDelete(TestBase):
+    """Nova Deferred Delete test cases"""
+
+    @classmethod
+    @pytest.fixture(scope='class', autouse=True)
```

Comment:
Now a see the difference.
WITHOUT @pytest.mark.undestructive:
```
$ py.test --pdb -s -v -ra --snapshot=ha_deploy_VLAN_CINDER --env=akor_Test_Deployment_MOS_CI mos_tests/nova/nova_test.py::TestNovaDeferredDelete
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_1
    mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    mos_tests.nova.nova_test:616: ~~ SETUP ~~
    mos_tests.nova.nova_test:625: test_1
    PASSED
    mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_2
    mos_tests.nova.nova_test:627: test_2
    PASSED
    mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_3
    mos_tests.nova.nova_test:629: test_3
    PASSED
    mos_tests.nova.nova_test:622: ~~ TEARDOWN ~~
```
WITH @pytest.mark.undestructive:
```
$ py.test --pdb -s -v -ra --snapshot=ha_deploy_VLAN_CINDER --env=akor_Test_Deployment_MOS_CI mos_tests/nova/nova_test.py::TestNovaDeferredDelete
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_1 
    mos_tests.environment.devops_client:131: Reverting snapshot ha_deploy_VLAN_CINDER
    mos_tests.nova.nova_test:617: ~~ SETUP ~~
    mos_tests.nova.nova_test:626: test_1
    PASSED
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_2 
    mos_tests.nova.nova_test:628: test_2
    PASSED
    
    mos_tests/nova/nova_test.py::TestNovaDeferredDelete::test_3 
    mos_tests.nova.nova_test:630: test_3
    PASSED
    mos_tests.nova.nova_test:623: ~~ TEARDOWN ~~
```